File: C++/Step9-Math_2 Prime and Geometric/1085.cpp
//
//  1085.cpp
//  Step9-Math_2 Prime and Geometric
//
//  Created by 김예빈 on 2019. 9. 5..
//  Copyright © 2019년 김예빈. All rights reserved.
//

#include <iostream>
using namespace std;

int main(int argc, const char * argv[]) {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    
    int x, y, w, h, min_x, min_y;
    
    cin >> x >> y >> w >> h;
    
    min_x = w - x;
    min_y = h - y;
    
    if(min_x > x)
        min_x = x;
    
    if(min_y > y)
        min_y = y;
    
    if(min_x < min_y)
        cout << min_x;
    else
        cout << min_y;

    return 0;
}

File: BaekJoon Online Judge/Swift/Step09-Math_2 Prime and Geometric/1085.swift
//
//  1085.swift
//  Step09-Math_2 Prime and Geometric
//
//  Created by Yebin Kim on 2019/11/22.
//  Copyright © 2019 김예빈. All rights reserved.
//

import Foundation

let input = readLine()!.split(separator: " ").map{ Int($0)! }
let x = input[0]
let y = input[1]
let w = input[2]
let h = input[3]

var minX = w - x
var minY = h - y

if minX > x {
    minX = x
}

if minY > y {
    minY = y;
}

if minX < minY {
    print(minX)
} else {
    print(minY)
}
